src/features/os-config/index.ts

```diff
@@ -1,27 +1,45 @@
-import type { Application, RequestHandler } from 'express';
+import type { Application } from 'express';
 
 import {
 	DEVICE_CONFIG_OPENVPN_CA,
 	DEVICE_CONFIG_OPENVPN_CONFIG,
 	DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
+	LOGS_HOST,
 } from '../../lib/config';
 import { b64decode } from '../../lib/utils';
 
-const getOsConfiguration: RequestHandler = (_req, res) => {
-	res.json({
-		services: {
-			openvpn: {
-				config: DEVICE_CONFIG_OPENVPN_CONFIG,
-				ca: b64decode(DEVICE_CONFIG_OPENVPN_CA),
-			},
-			ssh: {
-				authorized_keys: DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
-			},
-		},
-		schema_version: '1.0.0',
-	});
+// OS service configurations
+const services = {
+	openvpn: {
+		config: DEVICE_CONFIG_OPENVPN_CONFIG,
+		ca: b64decode(DEVICE_CONFIG_OPENVPN_CA),
+	},
+	ssh: {
+		authorized_keys: DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
+	},
+};
+
+// Config.json migrations: changes should be evaluated for security risks before applying.
+const config = {
+	// config.json fields to add or change
+	update: {
```

Comment:
How about calling this field `overrides`?

Reply:
I like this suggestion! I'm modifying the improvement to move to a target-state based approach for API directives to os-config, and will use this as well as `null / None` for fields to be deleted. In the meantime, and before the arch call, I'll move this back to draft status temporarily. Thank you for looking at the code @thgreasi !

Reply:
I've removed the update/delete directives in favor of `overrides`, and also removed the possibility of deleting fields from config.json.